pytential/qbx/utils.py

```diff
@@ -382,6 +382,8 @@ def _make_centers(discr):
             2 * qbx_element_to_source_starts
             if not use_stage2_discr
             else None)
+    if qbx_element_to_center_starts is not None:
+        assert qbx_element_to_center_starts.dtype == qbx_element_to_source_starts
```

Comment:
```suggestion
        assert qbx_element_to_center_starts.dtype == qbx_element_to_source_starts.dtype
```
?

Reply:
That's weird. How did this ever pass then?

Reply:
Ha.. no idea :confused:

Reply:
Oh numpy. :facepalm: 

```python
>>> class A:
...    pass
... 
>>> a = A()
>>> a.dtype = np.dtype(np.int32)
>>> a.dtype == a
True
```